### PR TITLE
[PLU-102] feat: rich text editor input field

### PR DESCRIPTION
**Pull request by stanleynguyen**
## Problem

We have input fields that take HTML content, however, our current inputs are only able to do plain text. Hence, we have a two-fold problem:
- We have to massage the plaintext content produced by the plaintext editor, and if users decide to input raw HTML this content will be affected by the massaging that we do on the plaintext content as well
- Users have no easy way to preview the raw HTML content that they're putting in

## Solution

Implement a new type of input `rich-text` that uses tiptap

## Tests

All tests from #273 #274 #277 

## Deploy Notes

- [ ] Send emails to notify all users that might be affected if they decide to update the affected fields from their existing pipes

### CODE REVIEW

**Review by ogp-weeloong — APPROVED** [review]
LGTM

If possible, would be good to fix the variables popper obscuring the tiptap menu if window height is too small

**Review by ogp-weeloong — DISMISSED** [review]
wait wait I take back what I said, old pipes break even without editing because we are not translating `\n` to `<br>` anymore.

https://github.com/opengovsg/plumber/pull/277#discussion_r1396999735 

#278 needs to be merged in before this entire feature branch can be merged into `develop`.

**Comment by stanleynguyen** [comment]
> wait wait I take back what I said, old pipes break even without editing because we are not translating `\n` to `<br>` anymore.
> 
> [#277 (comment)](https://github.com/opengovsg/plumber/pull/277#discussion_r1396999735)
> 
> #278 needs to be merged in before this entire feature branch can be merged into `develop`.

Actually I wanna merge #278 after we have merged and release this, to limit the impact to just the email body field, so I've brought up the changes needed from #277

File: packages/frontend/src/components/RichTextEditor/ResizableImageExtension.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+      // If mouse is up, remove event listeners
+      if (!event.buttons) {
+        removeListeners()
+      }
+    }
+
+    window.addEventListener('mousemove', mouseMoveHandler)
+    window.addEventListener('mouseup', removeListeners)
+  }, [])
```
Comment (m0nggh) [inline packages/frontend/src/components/RichTextEditor/ResizableImageExtension.tsx]: need add missing dependency `updateAttributes`

File: packages/frontend/src/components/RichTextEditor/SuggestionPopper.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+import {
+  Button,
+  Collapse,
+  List,
+  ListItemButton,
+  ListItemText,
+  Paper,
+  Popper,
+  Typography,
+} from '@mui/material'
```
Comment (m0nggh) [inline packages/frontend/src/components/RichTextEditor/SuggestionPopper.tsx:12]: is this MUI gonna be changed in a subsequent PR?

File: packages/frontend/src/components/RichTextEditor/ResizableImageExtension.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+const MIN_WIDTH = 60
+const MAX_WIDTH = 750
+
+const ResizableImageTemplate = ({ node, updateAttributes }: NodeViewProps) => {
+  const imgRef = useRef<HTMLImageElement>(null)
+  const [resizingStyle, setResizingStyle] = useState<
+    Pick<CSSProperties, 'width'> | undefined
+  >()
+
+  const handleMouseDown = useCallback((event: React.MouseEvent) => {
```
Comment (m0nggh) [inline packages/frontend/src/components/RichTextEditor/ResizableImageExtension.tsx]: nit: can remove the `React.`

File: packages/frontend/src/components/RichTextEditor/SuggestionPopper.tsx (hunk 2 above)
Comment (stanleynguyen) [reply packages/frontend/src/components/RichTextEditor/SuggestionPopper.tsx:12]: No this one is intended. Approved by Stace jiejie alr - "As long as the component looks like ODS, it's ok"

File: packages/frontend/src/components/RichTextEditor/ResizableImageExtension.tsx (hunk 3 above)
Comment (stanleynguyen) [reply packages/frontend/src/components/RichTextEditor/ResizableImageExtension.tsx]: This one cannot:
- it's to differentiate btwn React MouseEvent vs native MouseEvent
- also linting will behave weirdly if we import MouseEvent interface directly from 'react' 🤔

**Review by m0nggh — APPROVED** [review]
okay other than the image not appearing on the editor for some links, think its good to go, lgtm

File: packages/frontend/src/components/NewsDrawer/NewsItemList.ts [hunk 4]
```diff
@@ -8,6 +8,15 @@ const IF_THEN_EXTERNAL_LINK =
   'https://guide.plumber.gov.sg/user-guides/actions/toolbox'
 
 export const NEWS_ITEM_LIST: NewsItemProps[] = [
+  {
+    date: '2023-11-30',
+    tag: NEW_FEATURE_TAG,
+    title: 'Rich text editor for email body',
+    details: dedent`
+      * Email body can now be formatted easily with our brand-new rich-text editor with elements like table, list, headings, images.
```
Comment (m0nggh) [inline packages/frontend/src/components/NewsDrawer/NewsItemList.ts]: eh small nit: `brand new`

File: packages/frontend/src/components/NewsDrawer/NewsItemList.ts [hunk 5]
```diff
@@ -8,6 +8,15 @@ const IF_THEN_EXTERNAL_LINK =
...
+      * There might be some backward-compatbility issues if you are editing old pipes that use raw HTML. Keep a look out for our emails for more details.
```
Comment (m0nggh) [inline packages/frontend/src/components/NewsDrawer/NewsItemList.ts]: and also `backward compatibility`, thanksss

**Pull request merged**